modules/activiti-engine/src/test/java/org/activiti/engine/test/api/event/ActivityEventsTest.java

```diff
@@ -657,9 +749,51 @@ public void testActivityMessageBoundaryEventsOnSubProcess() throws Exception {
     assertTrue(signalEvent.getCause() instanceof MessageEventSubscriptionEntity);
     cause = (MessageEventSubscriptionEntity) signalEvent.getCause();
     assertEquals("message_1", cause.getEventName());
-
+    
+    assertDatabaseEventPresent(ActivitiEventType.ACTIVITY_MESSAGE_WAITING);
     assertDatabaseEventPresent(ActivitiEventType.ACTIVITY_MESSAGE_RECEIVED);
   }
+  
+  /**
+   * Test events related to message events, called from the API.
+   */
+  @Deployment(resources = "org/activiti/engine/test/api/event/ActivityEventsTest.testActivityMessageBoundaryEventsOnSubProcess.bpmn20.xml")
+  public void testActivityMessageBoundaryEventsOnSubProcessForCancel() throws Exception {
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("messageOnSubProcess");
+    assertNotNull(processInstance);
+
+    Execution executionWithMessage = runtimeService.createExecutionQuery().activityId("boundaryMessageEventCatching").singleResult();
+    assertNotNull(executionWithMessage);
+
+    Task task = taskService.createTaskQuery().processInstanceId(processInstance.getId()).singleResult();
+    taskService.complete(task.getId());
+
+    assertEquals(2, listener.getEventsReceived().size());
+
+    // First, a message waiting event is expected
+    assertTrue(listener.getEventsReceived().get(0) instanceof ActivitiMessageEvent);
+    ActivitiMessageEvent messageEvent = (ActivitiMessageEvent) listener.getEventsReceived().get(0);
+    assertEquals(ActivitiEventType.ACTIVITY_MESSAGE_WAITING, messageEvent.getType());
+    assertEquals("boundaryMessageEventCatching", messageEvent.getActivityId());
+    assertEquals(executionWithMessage.getId(), messageEvent.getExecutionId());
+    assertEquals(executionWithMessage.getProcessInstanceId(), messageEvent.getProcessInstanceId());
+    assertEquals(processInstance.getProcessDefinitionId(), messageEvent.getProcessDefinitionId());
+    assertEquals("message_1", messageEvent.getMessageName());
+    assertNull(messageEvent.getMessageData());
+
+    // Second, a message cancelled event is expected
+    assertTrue(listener.getEventsReceived().get(1) instanceof ActivitiMessageEvent);
+    messageEvent = (ActivitiMessageEvent) listener.getEventsReceived().get(1);
+    assertEquals(ActivitiEventType.ACTIVITY_MESSAGE_CANCELLED, messageEvent.getType());
+    assertEquals("boundaryMessageEventCatching", messageEvent.getActivityId());
+    assertEquals(executionWithMessage.getId(), messageEvent.getExecutionId());
+    assertEquals(executionWithMessage.getProcessInstanceId(), messageEvent.getProcessInstanceId());
+    assertEquals(processInstance.getProcessDefinitionId(), messageEvent.getProcessDefinitionId());
+    assertEquals("message_1", messageEvent.getMessageName());
+    assertNull(messageEvent.getMessageData());
+
+    assertDatabaseEventPresent(ActivitiEventType.ACTIVITY_MESSAGE_WAITING);
```

Comment:
An assert ```assertDatabaseEventPresent(ActivitiEventType.ACTIVITY_MESSAGE_CANCELLED);``` is needed here (and other tests where message are checked if they exist in DB). The assert will file there is no ```ActivityMessageEventHandler``` for ```ACTIVITY_MESSAGE_CANCELLED``` events. Can you add a ```ActivityMessageEventHandler``` for the new event?